uaclient/system.py

```diff
@@ -113,6 +113,12 @@ def get_dpkg_arch() -> str:
     return out.strip()
 
 
+@lru_cache(maxsize=None)
+def get_machine_type() -> str:
+    out, _ = subp(["systemd-detect-virt"])
+    return out.strip()
```

Comment:
can we be sure this is always installed?
we might need something like
```suggestion
def get_machine_type() -> Optional[str]:
    try:
        out, _ = subp(["systemd-detect-virt"])
        return out.strip()
    except ProcessExecutionError:
        return None
```

Reply:
This is delivered by systemd, which should be around on all releases we support. But you are right, better be safe than sorry

Reply:
@orndorffgrant updated. I have used an empty string as a return to avoid changing the type check for the platform dict. But let me know if you disagree

Reply:
Looks good! I'll just tag allan for his opinion on the default value in a new comment thread